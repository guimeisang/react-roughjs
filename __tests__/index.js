import React from 'react'
import { mount, configure, render } from 'enzyme' // mount render 都是可以渲染子组件的；并且mount是实际的dom所以可以模拟交互；render效率是mount的两倍；

import Adapter from 'enzyme-adapter-react-16'
import ReactRough from '../src'

// Setup Enzyme Adapter
configure({ adapter: new Adapter() })

describe('ReactRough', () => {
	describe('Core', () => {
		it('should render properly', () => {
			const wrapper = render(<ReactRough width={200} height={400} />)
			expect(wrapper).toMatchSnapshot()
		})

		it('should render properly with children', () => {
			const wrapper = render(
				<ReactRough width={200} height={400}>
					<ReactRough.Circle />
				</ReactRough>
			)
			expect(wrapper).toMatchSnapshot()
		})
	})

	describe('Arc', () => {
		it('should render properly with props', () => {
			const wrapper = mount(
				<ReactRough width={200} height={400}>
					<ReactRough.Arc points={[50, 50, 80]} fill="red" />
				</ReactRough>
			)

			expect(wrapper).toMatchSnapshot()
		})
	})

	describe('Circle', () => {
		it('should render properly with props', () => {
			const wrapper = mount(
				<ReactRough width={200} height={400}>
					<ReactRough.Circle points={[50, 50, 80]} fill="red" />
				</ReactRough>
			)

			expect(wrapper).toMatchSnapshot()
		})
	})

	describe('Ellipse', () => {
		it('should render properly with props', () => {
			const wrapper = mount(
				<ReactRough width={200} height={400}>
					<ReactRough.Ellipse
						points={[10, 50, 150, 80]}
						fill="blue"
						stroke="red"
					/>
				</ReactRough>
			)

			expect(wrapper).toMatchSnapshot()
		})
	})

	describe('Line', () => {
		it('should render properly with props', () => {
			const wrapper = mount(
				<ReactRough width={200} height={400}>
					<ReactRough.Line
						points={[60, 60, 190, 60]}
						fill="blue"
						stroke="red"
					/>
				</ReactRough>
			)

			expect(wrapper).toMatchSnapshot()
		})
	})

	describe('Path', () => {
		it('should render properly with props', () => {
			const wrapper = mount(
				<ReactRough width={200} height={400}>
					<ReactRough.Path
						dataString="M80 80 A 45 45, 0, 0, 0, 125 125 L 125 80 Z"
						fill="blue"
						stroke="red"
					/>
				</ReactRough>
			)

			expect(wrapper).toMatchSnapshot()
		})
	})
})
