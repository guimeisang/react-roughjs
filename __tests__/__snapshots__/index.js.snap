// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactRough Arc should render properly with props 1`] = `
<ReactRough
  height={400}
  width={200}
>
    <canvas
    height={400}
    width={200}
  >
  <Component
      fill="red"
      points={
        Array [
          50,
          50,
          80,
        ]
      }
    >
    <RoughConsumer
        fill="red"
        points={
          Array [
            50,
            50,
            80,
          ]
        }
        type="arc"
      />
    </Component>
  </canvas>
</ReactRough>
`;

exports[`ReactRough Circle should render properly with props 1`] = `
<ReactRough
  height={400}
  width={200}
>
  <canvas
    height={400}
    width={200}
  >
    <Component
      fill="red"
      points={
        Array [
          50,
          50,
          80,
        ]
      }
    >
      <RoughConsumer
        fill="red"
        points={
          Array [
            50,
            50,
            80,
          ]
        }
        type="circle"
      />
    </Component>
  </canvas>
</ReactRough>
`;

exports[`ReactRough Core should render properly 1`] = `
<canvas
  height="400"
  width="200"
/>
`;

exports[`ReactRough Core should render properly with children 1`] = `
<canvas
  height="400"
  width="200"
/>
`;

exports[`ReactRough Ellipse should render properly with props 1`] = `
<ReactRough
  height={400}
  width={200}
>
  <canvas
    height={400}
    width={200}
  >
    <Component
      fill="blue"
      points={
        Array [
          10,
          50,
          150,
          80,
        ]
      }
      stroke="red"
    >
      <RoughConsumer
        fill="blue"
        points={
          Array [
            10,
            50,
            150,
            80,
          ]
        }
        stroke="red"
        type="ellipse"
      />
    </Component>
  </canvas>
</ReactRough>
`;


exports[`ReactRough Line should render properly with props 1`] = `
<ReactRough
  height={400}
  width={200}
>
  <canvas
    height={400}
    width={200}
  >
    <Component
      fill="blue"
      points={
        Array [
          60,
          60,
          190,
          60,
        ]
      }
      stroke="red"
    >
      <RoughConsumer
        fill="blue"
        points={
          Array [
            60,
            60,
            190,
            60,
          ]
        }
        stroke="red"
        type="line"
      />
    </Component>
  </canvas>
</ReactRough>
`;

exports[`ReactRough Path should render properly with props 1`] = `
<ReactRough
  height={400}
  width={200}
>
  <canvas
    height={400}
    width={200}
  >
    <Component
      dataString="M80 80 A 45 45, 0, 0, 0, 125 125 L 125 80 Z"
      fill="blue"
      stroke="red"
    >
      <RoughConsumer
        dataString="M80 80 A 45 45, 0, 0, 0, 125 125 L 125 80 Z"
        fill="blue"
        stroke="red"
        type="path"
      />
    </Component>
  </canvas>
</ReactRough>
`;
